### ISSUE TITLE
Cannot run tests locally

### ISSUE THREAD

**Issue by thorseye** [INSPIRE-MIF/helpdesk-validator#675]
Before reporting an issue with the INSPIRE validator deployed at JRC, please check the [list of known limitations](https://github.com/INSPIRE-MIF/helpdesk-validator/wiki/Known-limitations).

I'm running the docker image locally using 

```
docker run --name inspire-validator -d -p 8090:8090 -v ~/etf:/etf ghcr.io/inspire-mif/helpdesk-validator/inspire-validator:2021.3.1
```

The container starts fine, and I'm able to access the GUI on http://localhost:8090/validator

When I try to run a test, I get a `HTTP 403` with the following error:

```json
{
  "error": "Access to the private subnet was rejected by a configuration setting!",
  "timestamp": "1638950105263",
  "url": "http://localhost/validator/v2/TestRuns",
  "id": "l.rejected.private.subnet.access",
  "stacktrace": [
    "de.interactive_instruments.etf.webapp.controller.LocalizableApiError",
    "\tat de.interactive_instruments.etf.webapp.controller.TestObjectController.createWithUrlResources(TestObjectController.java:294)",
    "\tat de.interactive_instruments.etf.webapp.controller.TestObjectController.initResourcesAndAdd(TestObjectController.java:428)",
    "\tat de.interactive_instruments.etf.webapp.controller.TestRunController.start(TestRunController.java:497)",
    "\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)",
    "\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)",
    "\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)",
    "\tat java.lang.reflect.Method.invoke(Method.java:497)",
    "\tat org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)",
    "\tat org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:133)",
    "\tat org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:97)",
    "\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)",
    "\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)",
    "\tat org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)",
    "\tat org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:967)",
    " [wrapped] org.springframework.web.util.NestedServletException: Handler dispatch failed; nested exception is de.interactive_instruments.etf.webapp.controller.LocalizableApiError",
    "\tat org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:982)",
    "\tat org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)",
    "\tat org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)",
    "\tat org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872)",
    "\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:707)",
    "\tat org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)",
    "\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)",
    "\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:821)",
    "\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1685)",
    "\tat de.interactive_instruments.etf.webapp.filter.ApiFilter.doFilterInternal(ApiFilter.java:89)",
    "\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)",
    "\tat org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347)",
    "\tat org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263)",
    "\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1668)",
    "\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:581)",
    "\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)",
    "\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)",
    "\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:226)",
    "\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1158)",
    "\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511)",
    "\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)",
    "\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1090)",
    "\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)",
    "\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:213)",
    "\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:109)",
    "\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:119)",
    "\tat org.eclipse.jetty.server.Server.handle(Server.java:517)",
    "\tat org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:308)",
    "\tat org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:242)",
    "\tat org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:261)",
    "\tat org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)",
    "\tat org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:75)",
    "\tat org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceAndRun(ExecuteProduceConsume.java:213)",
    "\tat org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:147)",
    "\tat org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:654)",
    "\tat org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:572)",
    "\tat java.lang.Thread.run(Thread.java:745)"
  ]
}
```

I tried to update the configuration in the war file as suggested in https://github.com/INSPIRE-MIF/helpdesk-validator/issues/370, unfortunately without success. 

Would it be possible to get some advice on how to get the container working locally?

**Comment by fabiovinci** [comment]
Dear @thorseye,

do you still have the problem? Have you tried the latest version?

**Comment by jenriquesoriano** [comment]
Dear @thorseye , 

just to confirm we have tested it against v2022.1 release and we have not been able to reproduce the behaviour.
Thus, as @fabiovinci pointed, we recommend you to use this version.
Thanks a lot.

Best regards

**Comment by thorseye** [comment]
@fabiovinci @jenriquesoriano I've tested the latest version and it seems to work as expected! Nice work, thank you! 👍

**Comment by milanSch** [comment]
Hi, 
I have solved this problem (in versions 2022.0 and later) by changing parameter

etf.testobject.allow.privatenet.access = false 
to
etf.testobject.allow.privatenet.access = true


It is in container directory in config file:
root@myserver: ... /etf/config# vi etf-config.properties

\# Allow access to private networks. Checked when
\# a new webservice test object is created. "false" is recommended in public
\# deployments.
\# Default: false
\# etf.testobject.allow.privatenet.access = false
etf.testobject.allow.privatenet.access = true



And restart container.

milan
